catalogs/catalog-generic-lakehouse/src/main/java/org/apache/gravitino/catalog/lakehouse/lance/LanceDataTypeConverter.java

```diff
@@ -19,174 +19,168 @@
-      switch (floatingPoint.getPrecision()) {
-        case SINGLE:
-          return Types.FloatType.get();
-        case DOUBLE:
-          return Types.DoubleType.get();
-        default:
-          throw new UnsupportedOperationException(
-              "Unsupported FloatingPoint precision: " + floatingPoint.getPrecision());
-      }
-    } else if (arrowType instanceof ArrowType.FixedSizeBinary) {
-      ArrowType.FixedSizeBinary fixedSizeBinary = (ArrowType.FixedSizeBinary) arrowType;
-      return Types.FixedType.of(fixedSizeBinary.getByteWidth());
-    } else if (arrowType instanceof ArrowType.Date) {
-      return Types.DateType.get();
-    } else if (arrowType instanceof ArrowType.Time) {
-      return Types.TimeType.get();
-    } else if (arrowType instanceof ArrowType.Timestamp) {
-      return Types.TimestampType.withoutTimeZone();
-    } else if (arrowType instanceof ArrowType.Utf8) {
-      return Types.StringType.get();
-    } else if (arrowType instanceof ArrowType.Binary) {
-      return Types.BinaryType.get();
-      // TODO handle complex types like List, Map, Struct
-    } else {
-      throw new UnsupportedOperationException("Unsupported Arrow type: " + arrowType);
-    }
-  }
-
-  public List<ArrowType> getChildTypes(Type parentType) {
-    if (parentType.name() != Type.Name.UNPARSED) {
-      return List.of();
-    }
-
-    List<ArrowType> arrowTypes = Lists.newArrayList();
-    String typeStr = ((UnparsedType) parentType).unparsedType().toString();
-    try {
-      Type t = JsonUtils.anyFieldMapper().readValue(typeStr, Type.class);
-      if (t instanceof Types.ListType listType) {
-        arrowTypes.add(fromGravitino(listType.elementType()));
-      } else if (t instanceof Types.MapType mapType) {
-        arrowTypes.add(fromGravitino(mapType.keyType()));
-        arrowTypes.add(fromGravitino(mapType.valueType()));
-      } else {
-        // TODO support struct type.
-        throw new UnsupportedOperationException(
-            "Unsupported UnparsedType conversion: " + t.simpleString());
-      }
-
-      return arrowTypes;
-    } catch (Exception e) {
-      // FixedSizeListArray(integer, 3)
-
-      try {
-        if (typeStr.startsWith("FixedSizeListArray")) {
-          String type = typeStr.substring(typeStr.indexOf('(') + 1, typeStr.indexOf(',')).trim();
-          Type childType = JsonUtils.anyFieldMapper().readValue("\"" + type + "\"", Type.class);
-          arrowTypes.add(fromGravitino(childType));
-
-          return arrowTypes;
-        }
-      } catch (Exception e1) {
-        throw new UnsupportedOperationException("Failed to parse UnparsedType: " + typeStr, e1);
-      }
-
-      throw new UnsupportedOperationException("Failed to parse UnparsedType: " + typeStr, e);
-    }
+    // since the table metadata will load from Gravitino storage directly, we don't need to
+    // implement this method for now.
+    throw new UnsupportedOperationException("toGravitino is not implemented yet.");
```

Comment:
Do we need such a conversion when request is sending from Lance rest server to Gravitino?

Reply:
Yes, I will implement this method when I refine the Lance REST server type conversion.